### PR TITLE
add backend/rtlsym.d

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/backend/rtlsym.d [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+ * Source:      $(DMDSRC backend/_rtlsym.d)
+ */
+
+module ddmd.backend.rtlsym;
+
+import ddmd.backend.cc : Symbol;
+
+enum
+{
+    RTLSYM_THROW,
```
Comment (andralex) [inline src/backend/rtlsym.d]: Are most of these enum values new? If so, please make them lowercase and replace the RTLSYM_ prefix with the name of the enum itself.

**Review by andralex — COMMENTED** [review]
This does not seem to be a mechanical change so let's make it right. Thanks!

File: src/backend/rtlsym.d (hunk 1 above)
Comment (WalterBright) [inline src/backend/rtlsym.d]: done

File: src/backend/rtlsym.d [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+module ddmd.backend.rtlsym;
+
+import ddmd.backend.cc : Symbol;
+
+enum RTLSYM
+{
+    throwd,
+    throwc,
+    throwdwarf,
+    monitor_handler,
```
Comment (andralex) [inline src/backend/rtlsym.d]: So... sorry these are still up to snuff. There seems to be a mix of two conventions here: (a) rawconcatenation, (b) things_with_underscores. Neither is correct. The convention we currently use at least for druntime and phobos is camelCase.

File: src/backend/rtlsym.d [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     backendlicense.txt
+ * Source:      $(DMDSRC backend/_rtlsym.d)
+ */
+
+module ddmd.backend.rtlsym;
+
+import ddmd.backend.cc : Symbol;
+
+enum RTLSYM
```
Comment (andralex) [inline src/backend/rtlsym.d]: RTLSym please

File: src/backend/rtlsym.d (hunk 2 above)
Comment (WalterBright) [inline src/backend/rtlsym.d]: Keep in mind that this is a translation from C++ code to D. The idea in translations is to change as little as practical, in order to make finding problems with the translation much, much easier.

Refactoring (and that includes changing naming conventions) comes *after*. I have a lot of experience translating code from language to language, and have learned this lesson the hard way repeatedly.

With these requested changes, I am losing greppability between the C++ and D sources, which will make dev work harder, not easier. Renaming should happen only after **all** the C++ usages have been converted.

I'm going to revert this back to the way it was originally submitted.

Comment (WalterBright) [inline src/backend/rtlsym.d]: Reversion done.

Comment (andralex) [inline src/backend/rtlsym.d]: I didn't know there is a corresponding C++ source with these names. Where is it?

Comment (WalterBright) [inline src/backend/rtlsym.d]: https://github.com/dlang/dmd/blob/master/src/backend/rtlsym.h

The full names are generated by macro expansion here:

https://github.com/dlang/dmd/blob/master/src/backend/cc.h#L1616

and are used in both the front end C++ source and the backend C++ source.

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**